In [1]:
import pandas as pd

In [2]:

spadl_h5 = "C:\\Users\\Arnau Climent\\OneDrive\\Documentos\\1_MASTER\\PORTFOLIO\\futbol-data-visualizacion\\archivos\\spadl-statsbomb.h5"
spadlstore = pd.HDFStore(spadl_h5, 'r') # 'r' indica modo lectura

In [4]:

spadlstore

<class 'pandas.io.pytables.HDFStore'>
File path: C:\Users\Arnau Climent\OneDrive\Documentos\1_MASTER\PORTFOLIO\futbol-data-visualizacion\archivos\spadl-statsbomb.h5

In [5]:
# Cargar la información de las competiciones
competitions_df = spadlstore['competitions']
print(competitions_df.head())

# Cargar la información de los partidos
games_df = spadlstore['games']
print(games_df.head())

# Cargar la información de los equipos
teams_df = spadlstore['teams']
print(teams_df.head())

# Cargar la información de los jugadores
players_df = spadlstore['players']
print(players_df.head())

# Cargar la información de la participación de los jugadores en los partidos
player_games_df = spadlstore['player_games']
print(player_games_df.head())

    season_id  competition_id competition_name country_name  \
38         90              11          La Liga        Spain   

   competition_gender season_name  
38               male   2020/2021  
   game_id  season_id  competition_id competition_stage  game_day  \
0  3773386         90              11    Regular Season         8   
1  3773565         90              11    Regular Season        18   
2  3773457         90              11    Regular Season        37   
3  3773631         90              11    Regular Season        22   
4  3773665         90              11    Regular Season        26   

            game_date  home_team_id  away_team_id  home_score  away_score  \
0 2020-10-31 21:00:00           206           217           1           1   
1 2021-01-09 18:30:00          1049           217           0           4   
2 2021-05-16 18:30:00           217           209           1           2   
3 2021-02-07 21:00:00           218           217           2           3   
4

In [8]:
game_id_ejemplo = 3773386 # Reemplaza con el ID del partido que te interese
actions_partido_df = spadlstore[f'actions/game_{game_id_ejemplo}']
print(actions_partido_df.head())

   game_id                     original_event_id  period_id  time_seconds  \
0  3773386  208f072d-cffa-4575-bd73-e9e9384c8fe1          1         0.689   
1  3773386  e1f11e14-fb24-4c82-868f-507385122b9c          1         1.653   
2  3773386  e5a0193d-1989-4c55-89e2-2f84c0a9ee7b          1         2.652   
3  3773386  4d801bc4-907a-4474-9620-7c6ca2610ea7          1         5.344   
4  3773386  9c9ef7ce-4d9d-4c9c-b5db-9c315e2947ed          1         6.608   

   team_id  player_id  start_x  start_y    end_x   end_y  type_id  result_id  \
0      217     5487.0  52.0625   33.660  61.6875  38.165        0          1   
1      217     5203.0  62.2125   37.740  64.2250  24.735        0          1   
2      217     8118.0  64.2250   24.735  63.3500  24.735       21          1   
3      217     8118.0  63.3500   24.735  74.3750  42.585        0          1   
4      217     5213.0  74.3750   42.585  66.6750  42.245       21          1   

   bodypart_id  action_id  
0            4          0  


Cada fila de este DataFrame representa una acción individual que ocurrió durante el partido con ID 3773386. Las columnas proporcionan información detallada sobre cada acción:

#### game_id: El identificador único del partido (en este caso, 3773386).

original_event_id: El identificador único del evento tal como fue registrado por la fuente de datos original (StatsBomb en este caso).

#### period_id: El período del partido en el que ocurrió la acción (1 para el primer tiempo, 2 para el segundo, etc.).

time_seconds: El tiempo transcurrido en segundos desde el inicio del período en el que ocurrió la acción.

#### team_id: El identificador único del equipo que realizó la acción. En este caso, todas las primeras acciones fueron realizadas por el equipo con team_id = 217.

player_id: El identificador único del jugador que realizó la acción. Vemos diferentes player_ids a lo largo de las primeras acciones.

#### start_x, start_y: Las coordenadas (en un sistema de coordenadas específico del proveedor de datos) del punto donde comenzó la acción.
end_x, end_y: Las coordenadas del punto donde terminó la acción.
#### type_id: El identificador del tipo de acción (por ejemplo, pase, tiro, falta, etc.). Necesitarás una tabla de mapeo para entender qué representa cada type_id.

result_id: El identificador del resultado de la acción (por ejemplo, éxito, fallo, fuera de juego, etc.). También necesitarás una tabla de mapeo para interpretar estos IDs.

#### bodypart_id: El identificador de la parte del cuerpo utilizada para realizar la acción (por ejemplo, pie derecho, pie izquierdo, cabeza, etc.). Se requiere una tabla de mapeo para su interpretación.

action_id: Un identificador secuencial de la acción dentro del partido (empezando desde 0).